src/storage-operators/src/persist_source.rs

```diff
@@ -983,7 +983,6 @@ mod tests {
     }
 
     #[mz_ore::test]
-    #[cfg_attr(miri, ignore)] // Reports undefined behavior
```

Comment:
im amazed this actually runs, but causes undefined issue? that may very well be in timely, but i don't know...do you mind filing an issue for this?

Reply:
It runs fine and didn't find any undefined behavior in my local run and in CI, so I'm not sure what that comment referred to. I got scared reading the comment which led to this PR.